docs/system_overview/configuration.md

```diff
@@ -923,3 +923,52 @@ $ iadmin get_grid_configuration authentication password_max_time
+
+### Client: `numThreads` keyword
+
+The way that the client can request the number of threads to use in parallel transfer is via the `numThreads` keyword. For some iCommands, this is provided using the `-N` option. From the [`iput` help text](../../icommands/user/#iput):
+```
+ -N  numThreads - the number of threads to use for the transfer. A value of
+       0 means no threading. By default (-N option not used) the server 
+       decides the number of threads to use.
+```
+
+### Legacy Static Policy Enforcement Point - `acSetNumThreads`
+
+The server configuration for number of parallel transfer threads can also be controlled through a **legacy** static policy enforcement point called `acSetNumThreads`. This policy can call `msiSetNumThreads` to configure the environment in such a way that the desired number of threads is used in parallel transfers. The policy is configured by default in `core.re` like this:
+```c
+acSetNumThreads {
+	msiSetNumThreads("default", "default", "default");
+}
+```
+It is recommended that the administrator leaves this PEP in its default implementation, providing "default" for the 3 parameters for `msiSetNumThreads`. This will allow the `default_number_of_transfer_threads` configuration in `server_config.json`'s `advanced_settings` to take effect as a default and a maximum.
+
+### Server: `server_config` Advanced Settings
+
+The `server_config.json` file for configuration of the iRODS server has a setting under `advanced_settings` called `default_number_of_transfer_threads`. This is the server's configuration of the number of threads to use by default in parallel transfers. If the client does not provide a number of threads to use for parallel transfer, this is the default value used in the other server-side configurations that will be explained in the following sections. Additionally, if `acSetNumThreads` is left in its default implementation (with "default" used for all parameters), the `default_number_of_transfer_threads` will also ensure that no client-provided number of transfer threads will exceed the value provided.
+
+The administrator may wish to disable parallel transfer threads altogether. This can be done by setting `default_number_of_transfer_threads` to 0 in the `advanced_settings`.
+
+It is recommended that careful consideration be given by the administrator when changing the default value. This will affect available resources on the configured server and could degrade performance if mis-configured. The iRODS Consortium researched the parallel transfer threads as it relates to performance and reached the default value in use today as a result. As such, it is recommended to leave the default value in place for most use cases. For more information on this topic, you can find the research paper here: [https://irods.org/uploads/2016/09/russell2016-iRODS-networking-419.pdf](https://irods.org/uploads/2016/09/russell2016-iRODS-networking-419.pdf)
```

Comment:
i'd like to nominate this blog post instead... https://irods.org/2016/09/irods-4-1-9-networking-performance-whitepaper/

Reply:
Done